File: lar-demo/leetcode/question/1446.连续字符.go

func maxPower(s string) int {
	res,count := 1,1

	for i := 1; i < len(s); i++ {
		if s[i] == s[i - 1] {
			count++
			if count > res {
				res = count
			}
		}else {
			count = 1
		}
	}
	return res
}

File: lar-demo/leetcode/question/1446.连续字符.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    //时间复杂度O(n)
    //空间复杂度O(1)
    function maxPower($s) {
        $res = $count = 1;

        for ($i = 1;$i < strlen($s);$i++)
        {
            if ($s[$i] == $s[$i - 1])
            {
                $count++;
                $res = max($res,$count);
            }else
            {
                $count = 1;
            }
        }
        return $res;
    }
}
